core/java/android/nfc/tech/NfcA.java

```diff
@@ -70,9 +70,7 @@ public NfcA(Tag tag) throws RemoteException {
         super(tag, TagTechnology.NFC_A);
         Bundle extras;
         mSak = 0;
-        if(tag.hasTech(TagTechnology.MIFARE_CLASSIC))
-        {
-            extras = tag.getTechExtras(TagTechnology.MIFARE_CLASSIC);
+        if (extras = tag.getTechExtras(TagTechnology.MIFARE_CLASSIC) != null) {
```

Comment:
Can't do this in java. Must be separated in 2 lines

Reply:
See force pushed :) Thank you

Reply:
You can do that in java, I swear! :P

Thanks!

Reply:
javac begs to differ:
`frameworks/base/core/java/android/nfc/tech/NfcA.java:73: error: incompatible types: boolean cannot be converted to Bundle
        if (extras = tag.getTechExtras(TagTechnology.MIFARE_CLASSIC) != null) {
                                                                     ^
frameworks/base/core/java/android/nfc/tech/NfcA.java:73: error: incompatible types: Bundle cannot be converted to boolean
        if (extras = tag.getTechExtras(TagTechnology.MIFARE_CLASSIC) != null) {
                   ^
2 errors
`

Reply:
I forgot some (). Sorry

if ((extras = tag.getTechExtras(TagTechnology.MIFARE_CLASSIC)) != null) {

That should work

Reply:
AHHHH I didn't even notice. Yea that should do. But oh well, Already pushed :) Should work the same.

Reply:
Yeah, no problem, thanks!